src/subset.rs

```diff
@@ -220,7 +227,22 @@ impl<N: NodeIdT + Rand> Subset<N> {
                 return Ok(step);
             }
         };
-        self.broadcast_results.insert(proposer_id.clone(), value);
+
+        let val_to_insert = if let Some(true) = self.ba_results.get(proposer_id) {
```

Comment:
No need to pattern-match here: This could just be `if Some(true) == self.ba_results.get(proposer_id)`.

On the other hand, maybe it should be a `match` instead: I'm wondering whether in the `Some(false)` case, we should also just insert `None`, since we won't need the value anyway—we already know it has been rejected?